### PR TITLE
feat: SfRating add support for fractions

### LINKED ISSUE #1232: [FEATURE] SfRating - support fractions
**Describe how this feature will help us improve**
Currently SfRating only supports fully integer scores. IMO supporting fractions like "4.7" is more common and used in eCommerce, so adding it would be extremaly useful.

**Is your proposal related to a problem?**
No

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue
Closes #1232 

# Scope of work
Add new SfIcon with linear gradient svg with proper fraction styling.

# Screenshots of visual changes

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed

### CODE REVIEW

**Comment by AdamPawlinski** [comment]
Fractional finalScore added, but I could not find solution other than importing icons similar to SfIcon. So now it is doubled.

**Comment by pspaczek** [comment]
@AdamPawlinski nice! Maybe we should add now props for SfIcon like `coverage`? 
```
coverage: {
type: [String, Number],
default: 1
}
```
if coverage < 1 then you can add defs with linearGradient

File: packages/vue/src/components/atoms/SfRating/SfRating.vue [hunk 1]
```diff
@@ -6,25 +6,61 @@
...
+          <path
+            v-for="(path, index) in iconPaths"
+            :key="index"
+            :d="path"
+            :fill="fillUrl(finalScore - Math.floor(finalScore))"
+            style="height: 100%;"
+          />
+        </svg>
+      </SfIcon>
+    </slot>
```
Comment (pspaczek) [inline packages/vue/src/components/atoms/SfRating/SfRating.vue:44]:
```suggestion
```
what do you think about use v-for for all positive icons and set right `coverage` for last with fractional?

File: packages/shared/styles/components/atoms/SfRating.scss [hunk 2]
```diff
@@ -3,8 +3,15 @@
   display: flex;
   &__icon {
     --icon-color: var(--c-primary);
+    --icon-color-negative: var(--c-gray-variant);
```
Comment (pspaczek) [inline packages/shared/styles/components/atoms/SfRating.scss:6]: @AdamPawlinski when you add coverage or fraction property you can add --icon-fraction in SfIcon scss and after that refactor this changes to adjust to these changes :)

File: packages/vue/src/components/atoms/SfRating/SfRating.vue (hunk 1 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/atoms/SfRating/SfRating.vue]: moved to SfIcon

File: packages/shared/styles/components/atoms/SfRating.scss (hunk 2 above)
Comment (AdamPawlinski) [reply packages/shared/styles/components/atoms/SfRating.scss:6]: fixes made

File: packages/shared/styles/components/atoms/SfIcon.scss [hunk 3]
```diff
@@ -17,6 +17,9 @@
     top: var(--icon-badge-top, 0);
     right: var(--icon-badge-right, 0);
   }
+  // &--fraction {
```
Comment (aniamusial) [inline packages/shared/styles/components/atoms/SfIcon.scss:20]: Should we keep it if its commented?

Comment (AdamPawlinski) [reply packages/shared/styles/components/atoms/SfIcon.scss:20]: Of course not, I forgot to remove it :)

**Pull request merged**